app/controllers/products_controller.rb

```diff
@@ -0,0 +1,56 @@
+class ProductsController < ApplicationController
+  before_action :authenticate_user!, except: [:show, :search, :index]
```

Comment:
do we need to check for published status

Reply:
For the scope of this pr I don't think so but I expect that will be needed for show page implementation